<launch>

  <!-- SLAM -->
  <include file="$(find lidar_slam)/launch/slam.launch"/>

  <!-- GPS conversion : Lat/Lon to UTM X/Y -->
  <!-- (WARNING depreciated) from sensor_msgs/NavSatFix to nav_msgs/Odometry -->
  <node name="navsatfix_to_utm" pkg="gps_common" type="utm_odometry_node" output="screen">
    <param name="frame_id" value="utm"/>
    <param name="child_frame_id" value="gps"/>
    <remap from="odom" to="gps_odom"/>
  </node>
  <!-- TODO from sensor_msgs/NavSatFix to gps_common/GPSFix -->
  <!-- from novariant_capi/PositionGps to gps_common/GPSFix -->
  <include file="$(find novariant_capi)/launch/gps_fix_conversion.launch"/>
  <!-- from gps_common/GPSFix to nav_msgs/Odometry -->
  <node name="gps_to_utm" pkg="gps_conversions" type="gps_to_utm" output="screen">
    <param name="frame_id" value="utm"/>
    <param name="child_frame_id" value="gps"/>
    <param name="map_frame_id" value="map"/>
    <param name="tf_offset_to_map" value="true"/>
    <param name="origin_on_first_pose" value="true"/>
  </node>

  <!-- SLAM to WGS84 conversion : UTM X/Y to GPS Lat/Lon -->
  <node name="slam_to_gps" pkg="gps_conversions" type="odom_to_gps" output="screen">
    <param name="utm_frame_id" value="utm"/>
    <remap from="odom" to="slam_odom"/>
    <remap from="fix" to="slam_fix"/>
  </node>

  <!-- Pose graph optimization using GPS and SLAM poses -->
  <node name="pose_graph_optimization" pkg="lidar_slam" type="pose_graph_optimization_node" output="screen">
    <rosparam file="$(find lidar_slam)/params/pose_graph_optim_config.yaml" command="load"/>
  </node>

</launch>